articles/concepts-and-configuration-of-stp/index.md

```diff
@@ -0,0 +1,111 @@
+### Introduction
+Loop prevention is one of the most important considerations when developing a switched network. In a switched network, once a loop forms, the amount of traffic that can be exchanged between switches easily consumes all of the bandwidth available within each of the affected switches. One way to avoid loops is to provide only a single path between switches and to [ensure that there is no path continuity in the switched network](https://www.ciscopress.com/articles/article.asp?p=1728837). The Spanning Tree Protocol is capable of averting loops in the switched network. It accomplishes this by temporarily obstructing redundant paths between switches. This article will go through some of the STP concepts as well as the configurations which are used to deploy STP on a switch.
+
+### Prerequisites 
+To follow through the article, you will need to:
+1. Have [Cisco packet tracer](https://www.netacad.com/courses/packet-tracer) installed.
+2. Have a basic knowledge of configuring [Cisco devices](https://www.networkstraining.com/basic-cisco-router-configuration-steps/).
+3. Have an understanding of Cisco networking concepts.
+
+### Purpose of the spanning tree protocol (STP)
+Redundancy is an important aspect of the hierarchical architecture for avoiding network service failures and removing single points of failure. Physical paths must be added to redundant networks, but logical redundancy must also be considered. Users can access network services despite path interruption by providing alternative physical paths for data to traverse the network. 
+
+In a switched Ethernet network, redundant paths, on the other hand, can trigger logical and physical layer 2 loops. The Spanning Tree Protocol is a loop-prevention network protocol that [creates a loop-free Layer 2 topology thus allowing for redundancy](https://www.ciscopress.com/articles/article.asp?p=1728837). STP's main goal is to create a topology of all participating STP switches to create a loop-free switched network. The best loop-free path through the switched network is then [calculated using this topology information](https://www.ciscopress.com/articles/article.asp?p=1728837).
+
+### The spanning tree protocol's (STP) operations 
+A root switch election occurs when all switches behave as if they are the root switch at first and continue to do so until they receive traffic from a superior switch (switch priority determines). This is known as a root switch election. Another thing to bear in mind is that depending on the STP mode used, the network may have multiple root switches. Each VLAN has its STP case on Cisco switching equipment, and a root switch is [automatically elected for each VLAN; this mode is known as Per VLAN Spanning Tree Plus (PVST+)](https://www.cciein8weeks.com/ccie-rs-400-101-v5-1-written-exam-implement-and-troubleshoot-spanning-tree). Rapid PVST+ is used when RSTP is implemented.
```

Comment:
> (switch priority determines)

How is switch priority determined? Please add a line or two elaborating this so that beginners follow easily.